cpp/environment.yml

```diff
@@ -0,0 +1,115 @@
+name: cookbook-cpp
+channels:
+  - conda-forge
+dependencies:
+  - _libgcc_mutex=0.1=conda_forge
```

Comment:
small nit, but might want to make this environment.yml file more generic? (only listing the actual packages needed (and not all their dependencies), and without the build hashes)

If you created this from a local conda environment you have, you can also use `conda env export --from-history` to get such a cleaner version.

Reply:
That was intentional, although maybe a little odd for the python world :).  It is often done in JS (e.g. npm/yarn) where they keep both the `package.xml` (in this case `requirements.txt`) and `package-lock.xml`.  The lock file is used most of the time but the package file is used when you want to upgrade to the latest version of all dependencies.  The purpose is two-fold:

1. Don't allow CI to fail because a downstream package pushed an update
2. Speed up CI because there is no need to run the solver.

The downside then is that you aren't always running against the latest and greatest and so you might miss an integration issue but, since these are only used for building the site (e.g. I'm not building a library), I don't think it's a big deal.

Things are a little trickier it seems with python because dependencies can have multiple distributions (JS dependencies that require native support build on install time) but as long as we're only using this in CI then I think it should be ok.